File: ms-pacientes/src/main/java/com/piedraazul/mspacientes/model/PacienteFactory.java
package com.piedraazul.mspacientes.model;

import com.piedraazul.mspacientes.dto.PacienteDTO;

// ══════════════════════════════════════════════════════
// PATRÓN GOF: FACTORY METHOD
// Propósito: centralizar la creación de objetos Paciente
// a partir de un DTO, ocultando la lógica de construcción.
// HU-02 / HU-09 — Registro de pacientes
// ══════════════════════════════════════════════════════
public class PacienteFactory {

    public static Paciente crearDesdeDTO(PacienteDTO dto) {
        return new PacienteBuilder(
                dto.getNombre(),
                dto.getApellido(),
                dto.getNumeroDocumento(),
                dto.getTelefono(),
                dto.getGenero()
        )
                .fechaNacimiento(dto.getFechaNacimiento())
                .email(dto.getEmail())
                .direccion(dto.getDireccion())
                .eps(dto.getEps())
                .estado(dto.getEstado() != null ? dto.getEstado() : EstadoPaciente.ACTIVO)
                .build();
    }
}

File: backend/src/modules/pacientes/domain/paciente.factory.js
import { Paciente } from "./paciente.entity.js";
import { EstadoPaciente } from "./estado-paciente.js";

export class PacienteFactory {
  static crearDesdeDTO(dto) {
    return new Paciente({
      nombre: dto.nombre,
      apellido: dto.apellido,
      numeroDocumento: dto.numeroDocumento,
      fechaNacimiento: dto.fechaNacimiento,
      email: dto.email ?? null,
      telefono: dto.telefono,
      direccion: dto.direccion ?? null,
      eps: dto.eps ?? null,
      estado: EstadoPaciente.ACTIVO,
      genero: dto.genero
    });
  }
}